src/helpers/chartHelpers.tsx

```diff
@@ -3,45 +3,47 @@ import { isNumber } from "lodash";
 import { AreaChartSeries, ChartSeries, ChartThreshold, MultiSeriesBarChartOptions, MultiSeriesLineChartOptions } from "./chartOptions";
 import { ColorDefinition, resolveColor } from "./colors";
 import { LayoutContext } from "../components";
+import { TimeSeriesDataPoint } from '../components/presentational/TimeSeriesChart/TimeSeriesChart';
 
-export function createLineChartDefs(layoutContext: LayoutContext, gradientKey: string, series: ChartSeries[], options?: MultiSeriesLineChartOptions) {
+export function createLineChartDefs(
+    layoutContext: LayoutContext, 
+    gradientKey: string, 
+    series: ChartSeries[], 
+    options: MultiSeriesLineChartOptions | undefined,
+    dataKeys: string[],
+    data: TimeSeriesDataPoint[] ) {
 
     const colorOrDefault = (colorDefinition: ColorDefinition | undefined, defaultColor: string) => {
         return resolveColor(layoutContext.colorScheme, colorDefinition) ?? defaultColor;
     };
 
-    let nDomainMin: number | undefined = undefined;
-    let nDomainMax: number | undefined = undefined;
-
-    if (options) {
-        if (isNumber(options.domainMin)) {
-            nDomainMin = options.domainMin;
-        }
-        nDomainMax = options.domainMax;
-    }
-
-    const getPercent = function (numerator: number, domainMin: number, domainMax: number): number {
-        if ((domainMax - domainMin) > 0) {
-            return (numerator - domainMin) / (domainMax - domainMin) * 100;
+    const getPercent = function (numerator: number, min: number, max: number): number {
+        if ((max - min) > 0) {
+            return (numerator - min) / (max - min) * 100;
         }
         return 0;
     }
 
-    const createStopsFromThresholds = function (defaultLineColor: string, domainMin: number, domainMax: number, chartThresholds?: ChartThreshold[]) {
+    const createStopsFromThresholds = function (defaultLineColor: string, chartThresholds: ChartThreshold[] | undefined, index: number) {
         let stops: any[] = [];
         let lineColor: string = defaultLineColor;
         let thresholds = chartThresholds ?? [];
 
         thresholds.sort((a, b) => a.value - b.value);
 
-        if (thresholds.length && domainMin >= thresholds[0].value) {
+        const lineRange = data.reduce((result, dataPoint) => ({
+            min: (dataPoint[dataKeys[index]] < result.min || result.min === 0) ? dataPoint[dataKeys[index]] : result.min,
+            max: (dataPoint[dataKeys[index]] > result.max || result.max === 0) ? dataPoint[dataKeys[index]] : result.max,
+        }), { min: 0, max: 0 });
```

Comment:
I think the `min` and `max` may need to be initialized to `undefined` and then checked for that in the `||` conditions.

Otherwise, I think if the `min` value is ever correctly set to `0` throughout the loop, the next data point will set the new minimum, even if it is greater than the correct minimum of `0`.

An alternative might be to do something like this:
```
let values = data.map(dataPoint => dataPoint[dataKeys[index]]);
let lineRange = {
	min: Math.min(...values),
	max: Math.max(...values)
}
```

Reply:
Wow, great catch.  I've fixed this by removing the explicit check against 0 as an initial value, and replaced the initial value with the first value in the data set.